frontend/src/ui/Blocks/Series.tsx

```diff
@@ -125,25 +125,49 @@ const ReadySeriesBlock: React.FC<ReadyProps> = ({
     const finalTitle = title ?? (showTitle ? series.title : undefined);
 
     const events = series.events.filter(event =>
-        !isPastLiveEvent(event.syncedData?.endTime ?? null, event.isLive));
+        !isPastLiveEvent(event.syncedData?.endTime ?? null, event.isLive)
+        && !isUpcomingEvent(event.syncedData?.startTime ?? null));
+
+    const upcomingEvents = series.events.filter(event =>
+        isUpcomingEvent(event.syncedData?.startTime ?? null));
 
     const sortedEvents = [...events];
-    sortedEvents.sort(match(order, {
-        "NEW_TO_OLD": () => compareNewToOld,
-        "OLD_TO_NEW": () => compareOldToNew,
-    }, unreachable));
+    sortedEvents
+        .sort(match(order, {
+            "NEW_TO_OLD": () => compareNewToOld,
+            "OLD_TO_NEW": () => compareOldToNew,
+        }, unreachable))
+        .sort((a, b) => +b.isLive - +a.isLive);
```

Comment:
Are you aware that this only works because `sort` uses a stable sorting algorithm? 
I think I would rather only call `sort` once. Instead of this whole `match` thing you can just code an arrow function in there, something like this:

```suggestion
    const timeMs = (event: Event) => new Date(event.syncedData?.startTime ?? event.created).getTime();
    sortedEvents.sort((a, b) => {
        // Sort all live events before non-live events.
        if (a.isLive !== b.isLive) {
            return +b.isLive - +a.isLive;
        }
        
        return match(order, {
            "NEW_TO_OLD": () => timeMs(a) - timeMs(b),
            "OLD_TO_NEW": () => timeMs(b) - timeMs(a),
        }, unreachable);
    })
```

Then, the two helper functions `compareNewToOld` and its friend can be deleted I think.

Reply:
I just assumed that it was stable. Just found out that it's apparently dependent on the browser's implementation. Interesting. I believe most browsers do implement it as stable, but that wasn't always the case and we should not rely on that. Thank you for the suggestion 👍

Reply:
Oh I actually assumed that the JS standard now guarantees that `sort` is stable :D 
I was rather commenting because the approach seemed rather unusual to me. I.e. sorting everything twice and relying on stable sorting to make *lexicographic sorting* happen. 
I see you changed it, so I'm resolving.